Changes needed to support Calibra ticket (regarding SiteDataPlotter)

ZDu-IDM opened the pull request:
This PR addressed the following

Add utility tools to convert Windows file path to Linux file path for Slurm #1659
https://github.com/InstituteforDiseaseModeling/idmtools/issues/1659

SiteDataPlotter needs to be refactored to support SlurmPlatform #157
https://github.com/InstituteforDiseaseModeling/idmtools-calibra/issues/157

**Note**: this PR did all the required idmtools change for above calibra ticket.

Refer to PR in calibra
**_PR:_** Refactoring SiteDataPlotter to support LL for both COMPS and SLURM platform #165
https://github.com/InstituteforDiseaseModeling/idmtools-calibra/pull/165

devclinton commented:
@ZDu-IDM - Can you fix the linting issues?

ZDu-IDM commented:
> @ZDu-IDM - Can you fix the linting issues?

Got all linting issues fixed, thanks!

shchen-idmod commented:
This PR failed lot of tests: https://github.com/shchen-idmod/idmtools-1/actions/runs/3425970433/jobs/5707286888 with same reason. I suspect, we may change environment variable in host somehow with this PR.

shchen-idmod commented on idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py:
```diff
@@ -467,3 +467,19 @@ def platform_list_asset(self, experiment: Experiment, **kwargs) -> List[Asset]:
+            experiment_id: experiment id
+
+        Returns:
+            Dict
```
Can Dict comment more specific?

ZDu-IDM commented:
> This PR failed lot of tests: https://github.com/shchen-idmod/idmtools-1/actions/runs/3425970433/jobs/5707286888 with same reason. I suspect, we may change environment variable in host somehow with this PR. Updated, tests are running on this branch as base and new test I added: [ZDu-IDM#1](https://github.com/ZDu-IDM/zdu_idmtools/pull/1). Failed tests seem appear after running my new testcases: test_create_sim_directory_map.py

[FIXED] make set/clear a pair usage, please verify, thanks!

shchen-idmod commented on idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py:
```diff
@@ -467,3 +467,19 @@ def platform_list_asset(self, experiment: Experiment, **kwargs) -> List[Asset]:
+        set_linux_mounts(self.platform)
+        comps_exp = self.platform.get_item(experiment_id, ItemType.EXPERIMENT, raw=True)
+        comps_sims = comps_exp.get_simulations(QueryCriteria().select(['id', 'state']).select_children('hpc_jobs'))
+        clear_linux_mounts(self.platform)
```
This line need after result dict.

ZDu-IDM replied:
[Fixed] Good catch, thanks!

shchen-idmod reviewed (approved):
All tests passed. approved~
